src/main/java/com/rtb/projectmanagementtool/private-comments/PrivateCommentController.java

```diff
@@ -0,0 +1,105 @@
+package com.rtb.projectmanagementtool.privatecomment;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.*;
+import com.rtb.projectmanagementtool.user.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Class controlling the PrivateCommentData object. */
+public final class PrivateCommentController {
+
+  private DatastoreService datastore;
+  private static final Filter NO_QUERY_FILTER = null;
+  private static final int NO_QUERY_LIMIT = Integer.MAX_VALUE;
+  private static final SortPredicate NO_QUERY_SORT = null;
+
+  public PrivateCommentController(DatastoreService datastore) {
+    this.datastore = datastore;
+  }
+
+  public Long addPrivateComment(PrivateCommentData privateComment) {
```

Comment:
If we're only allowing for 1 private comment per task, is there anyway we can enforce that?

Reply:
Not sure. Any recommendations?

Reply:
@Tzirliya thoughts?

Reply:
Maybe add a method that queries and filters for a privateComment with the current userID / taskID combination. If it returns null, create a new privateCommentData and add it to datastore, otherwise just update the retrieved entity.